### PR TITLE
Check execute_on option in the conservative transfer

### LINKED ISSUE #17186: Remove unused temperature from TensorMechanics action
## Reason
I'm _fairly_ certain that the `temperature` input parameter for the `TensorMechanicsActionBase` isn't actually used anywhere in that class or its derived classes. As a very new users, this is somewhat confusing as to its purpose (since all the tests still pass a `temperature` in to avoid `--no-warning` failures).

## Design
Remove the `temperature` parameter from `TensorMechanicsActionBase` and update any tests to remove the extraneous info.

## Impact
More straightforward understanding of the action system.

### LINKED ISSUE #17231: Check execute on option in the conservative transfer
## Reason

The conservative transfer requires users to have `execute_on=transfer` in the input file. But it is easy for users to forget. A moose-based application might hardcode `execute_on` options, and in this case, users' input file options will be ignored, leading to a wrong conversation transfer.

## Design

We check if users have execute_on=`transfer` in the input file. We will error out if users forget this option when they want to use the conservative transfer.

## Impact
Better user experience

### PR DESCRIPTION
**Pull request by fdkong**
Closes #17231

### CODE REVIEW

File: framework/src/problems/FEProblemBase.C [hunk 1]
```diff
@@ -3805,6 +3805,14 @@ FEProblemBase::joinAndFinalize(TheWarehouse::Query query, bool isgen)
   }
 }
 
+void
+FEProblemBase::computeUserObjectByName(const std::string & name)
+{
+  TheWarehouse::Query query =
+      theWarehouse().query().condition<AttribSystem>("UserObject").condition<AttribName>(name);
+  computeUserObjectsInternal(EXEC_NONE, Moose::POST_AUX, query);
```
Comment (lindsayad) [inline framework/src/problems/FEProblemBase.C:3813]: `POST_AUX` here seems pretty arbitrary

Comment (lindsayad) [reply framework/src/problems/FEProblemBase.C:3813]: Can you add an argument to this new API that can control the second argument to `computeUserObjectsInternal`?

**Review by lindsayad — CHANGES_REQUESTED** [review]
So this allows someone to compute a user object regardless of where you are in the simulation (e.g. regardless of whether I'm computing a residual, Jacobian, at the beginning of a timestep, etc.)? This seems like it could be dangerous. What if a user object only works at a particular stage of the simulation? I don't actually even know what `EXEC_NONE` means

**Comment by fdkong** [comment]
@lindsayad It is ready for review again

File: framework/src/transfers/MultiAppConservativeTransfer.C [hunk 2]
```diff
@@ -147,6 +147,51 @@ MultiAppConservativeTransfer::initialSetup()
...
+    {
+      // Get out all execute_on options for master pp
+      auto & execute_on = master_problem.getUserObjectBase(master_pp).getExecuteOnEnum();
+      // Check if master app has transfer execute_on
+      if (!execute_on.contains(EXEC_TRANSFER))
+        mooseError("execute_on='transfer' is required in the conversative transfer for ",
+                   master_pp,
+                   ".\n"
+                   "Please add execute_on='transfer' into input file, and also \n"
+                   "make sure that execute_on options are not hardcoded in the code \n");
```
Comment (lindsayad) [inline framework/src/transfers/MultiAppConservativeTransfer.C:168]: Love it!

**Review by lindsayad — APPROVED** [review]
Beautiful changes, beautiful code

**Review by aeslaughter — APPROVED** [review]
I like it!

**Pull request merged**